Code/Core/FileIO/FileIO.cpp

```diff
@@ -222,6 +223,20 @@
             return false;
         }
         linkPath.SetLength( length );
+
+        if (linkPath.Get()[0] != '/' )
+        {
+            // Convert linkPath to a full path in case the relative path
+            // is wrong from dstFileName
+            if ( srcFileName[0] != '/' )
```

Comment:
This was always true in my tests; I believe it's guaranteed that fbuild has expanded `srcFileName` and `dstFileName` to absolute paths by this point?  Need advice on what, if any, check is appropriate here and what to do if the check fails.

Reply:
The higher level code during the build deals with full paths in most cases, but this lower level code shouldn't assume the path passed in is a full path.